### PR TITLE
update camlp5 >= 8.0

### PR DESCRIPTION
**Pull request by gares**
TODO:
- [x] avoid workaround on OSX for missing depext instructions on perl stuff

### CODE REVIEW

**Comment by gares** [comment]
@chetmurthy I finally managed to make it work but I can't really merge this PR since we use elpi in the Coq platform, and there we need al non-ocaml dependencies to be handled by opam's depext mechanism. Here I had to invoke cpan by hand on OSX in order to get the new camlp5 conf-perl-* dependencies satisfied.

Could you fix the depext for OSX?

**Comment by Blaisorblade** [comment]
Could we get this in as a package in `extra-dev` as a compromise?

**Comment by gares** [comment]
I understand the pain, and I wont complain if you do put this branch out there. But I'd rather use my time on the branch adt/parser and ditch camlp5 altogether.

**Comment by chetmurthy** [comment]
> Could you fix the depext for OSX?

I have no access to MacOS machines.  I've been looking for a Mac user who knows enough about these distros (macports, homebrew) so that I could work with them to fix these depexts, but have been unable to find anybody with the requisite knowledge.  It's a simple thing to do, and if I had Mac hardware, it'd already be done.  Ah, well.

**Comment by k4rtik** [comment]
If you are still looking I won't mind assisting with homebrew-based testing. I have never used macports, however.

**Comment by chetmurthy** [comment]
> If you are still looking I won't mind assisting with homebrew-based testing. I have never used macports, however.

if this is directed to me, I'm "Chet Murthy" at ocaml.org: can you DM me there ?

**Comment by chetmurthy** [comment]
It should be resolved, yes.  Disabled testing on homebrew, fixed depext on macports. Should be resolved in currently-released camlp5.

**Comment by Blaisorblade** [comment]
Last I installed camlp5, the changes seemed to have been reverted... I'll take a look.

**Comment by chetmurthy** [comment]
version 8.00.03 should have the disabling of tests on homebrew.

**Comment by Blaisorblade** [comment]
Am I doing something wrong?
```
$ opam update; opam info camlp5.8.00.03
[...]
depends      "ocaml" {>= "4.05" & < "4.15.0"}
             "ocamlfind"
             "camlp-streams" {>= "5.0"}
             "conf-perl"
             "conf-perl-ipc-system-simple"
             "conf-perl-string-shellquote"
             "conf-diffutils" {with-test & os-distribution = "alpine"}
             "pcre" {with-test}
             "ounit" {with-test}
             "rresult" {with-test}
             "fmt" {with-test}
$ opam repo
[NOTE] These are the repositories in use by the current switch. Use '--all' to see all configured repositories.

<><> Repository configuration for switch 4.10.2+flambda <><><><><><><><><><>  🐫
 1 iris-dev     git+https://gitlab.mpi-sws.org/iris/opam.git
 2 default      https://opam.ocaml.org
 3 coq-released https://coq.inria.fr/opam/released
```

**Comment by chetmurthy** [comment]
that's right: the two conf-perl-* deps don't have depexts on brew.  Did the build go thru ?

**Comment by Blaisorblade** [comment]
No:

```
$ opam install conf-perl-ipc-system-simple conf-perl-string-shellquote
#=== ERROR while compiling conf-perl-string-shellquote.2 ======================#
# context     2.1.2 | macos/arm64 | ocaml-variants.4.10.2+flambda | https://opam.ocaml.org#00706137
# path        ~/.opam/4.10.2+flambda/.opam-switch/build/conf-perl-string-shellquote.2
# command     ~/.opam/opam-init/hooks/sandbox.sh build perl -MString::ShellQuote -e 1
# exit-code   2
# env-file    ~/.opam/log/conf-perl-string-shellquote-59775-657ea3.env
# output-file ~/.opam/log/conf-perl-string-shellquote-59775-657ea3.out
### output ###
# Can't locate String/ShellQuote.pm in @INC (you may need to install the String::ShellQuote module) (@INC contains: /opt/homebrew/Cellar/perl/5.34.0/lib/perl5/site_perl/5.34.0/darwin-thread-multi-2level /opt/homebrew/Cellar/perl/5.34.0/lib/perl5/site_perl/5.34.0 /opt/homebrew/Cellar/perl/5.34.0/lib/perl5/5.34.0/darwin-thread-multi-2level /opt/homebrew/Cellar/perl/5.34.0/lib/perl5/5.34.0 /opt/ho[...]
# BEGIN failed--compilation aborted.
```

your original fix did work, but it skipped the conf packages altogether.

**Comment by chetmurthy** [comment]
FWIW, you're the first person to test this on brew, so I guess we shouldn't be surprised that it didn't work.  I have no access to mac hardware, and the CI env didn't test it either.

**Comment by Blaisorblade** [comment]
No worries, I should have reported it when I saw it!

**Comment by chetmurthy** [comment]
I submitted a PR for this: https://github.com/ocaml/opam-repository/pull/20963

**Comment by Blaisorblade** [comment]
Thanks! Looks like it should fix this problem, but it's at least interesting that the solution is to let those two `conf-perl` packages be installed anyway, instead of saying that on homebrew we leave out the dependency. I don't think there's a concrete problem until somebody else depends on those.

**Comment by chetmurthy** [comment]
I don't remember *exactly* why we did it that way, but my memory is that the reviewer suggested it.  I think the idea was that someday, somebody with "homebrew" experience would figure out how to install those Perl packages on homebrew, and then everything would just work.

Now camlp5 works, but none of the follow-on packages (all the "pa_ppx*" packages, as well as other stuff I've built on camlp5) will work on homebrew, b/c they actually use those Perl packages for scripts used in building.  OTOH, they work on macports (IIRC).

**Comment by chetmurthy** [comment]
The fixes got merged, so soon you should be able to fetch and install from opam.  Could you let me know when you do, and if they work ?

The versions are:
conf-perl-ipc-system-simple.3
conf-perl-string-shellquote.3

**Comment by Blaisorblade** [comment]
That worked! `opam install conf-perl-ipc-system-simple conf-perl-string-shellquote` succeeded here and picked version 3, and I installed `elpi` 1.14.1 using a trivial forward-port of this branch (https://github.com/Blaisorblade/elpi/tree/update-camlp5-port).

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -59,12 +62,20 @@ jobs:
           opam exec -- sed -i ' ' tests/sources/*.elpi
           & "$Env:CYGWIN_ROOT/setup-x86_64.exe" -q -P time
 
+      - name: Satisfy depext of camlp5 on OSX workaround
+        if: runner.os == 'macOS'
+        run: |
+          cpan install IPC::System::Simple
+          cpan install String::ShellQuote
+
```
Comment (Blaisorblade) [inline .github/workflows/main.yml:70]: With @chetmurthy 's changes, we should be able to drop this block:
```suggestion
```

@gares I remember you have other plans, but I hope you'd be willing to try it.

**Comment by MSoegtropIMC** [comment]
@gares : is this something which could go in the next few days? It is blocking updating Coq Platform to OCaml 4.13.1.

**Comment by gares** [comment]
Looks green, I'll make a release later today

**Pull request merged**